### PR TITLE
FOUNDATIONS: Retrieve All Posts

### PR DESCRIPTION
**Pull request by stasnowak**
#25

### CODE REVIEW

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+namespace Taarafo.Portal.Web.Tests.Unit.Services.Foundations.Posts
+{
+    public partial class PostServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalDependencyExceptions))]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveAllIfCriticalDependencyExceptionOccursAndLogItAsync(
+            Exception criticalDependencyException)
+        {
+            var failedPostDependencyException =
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:25]: // given
// when
// then
is missing

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedPostDependencyException))),
+                        Times.Once);
+
+            this.apiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowPostDependencyExceptionOnRetrieveAllIfDependencyApiErrorOccursAndLogItAsync()
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:56]: We can remove Post from the test name as we are in the Post test scope.

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        {
+            var randomExceptionMessage = GetRandomMessage();
+            var responseMessage = new HttpResponseMessage();
+
+            var httpResponseException =
+                new HttpResponseException(
+                    httpResponseMessage: responseMessage,
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:63]: No need for aliasing as the parameter name is part of the actual name,

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+            await Assert.ThrowsAsync<PostDependencyException>(() =>
+                retrieveAllPostsTask.AsTask());
+
+            this.apiBrokerMock.Verify(broker =>
+                broker.GetAllPostsAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:88]: Why do we need Validation exception here?

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Taarafo.Portal.Web/Services/Foundations/Posts/PostService.cs [hunk 5]
```diff
@@ -24,6 +25,15 @@ public partial class PostService : IPostService
             this.loggingBroker = loggingBroker;
         }
 
+        public ValueTask<List<Post>> RetrieveAllPostsAsync() =>
+        TryCatch(async () =>
+        {
+            List<Post> posts =
+                await apiBroker.GetAllPostsAsync();
+
+            return posts;
```
Comment (hassanhabib) [inline Taarafo.Portal.Web/Services/Foundations/Posts/PostService.cs:34]: Why not just `async () => await this.apiBroker.GetAllPostsAsync());`

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+using Taarafo.Portal.Web.Models.Posts.Exceptions;
+using Xunit;
+
+namespace Taarafo.Portal.Web.Tests.Unit.Services.Foundations.Posts
+{
+    public partial class PostServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalDependencyExceptions))]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveAllIfCriticalDependencyExceptionOccursAndLogItAsync(
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:22]: Going beyond 120

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            await Assert.ThrowsAsync<PostDependencyException>(() =>
+                retrieveAllPostsTask.AsTask());
+
+            this.apiBrokerMock.Verify(broker =>
+                broker.GetAllPostsAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                        expectedDependencyException))),
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Exceptions.RetrieveAll.cs:95]: Remove extra tab space

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.cs [hunk 8]
```diff
@@ -48,6 +52,14 @@ public PostServiceTest()
                 (actualException.InnerException as Xeption).DataEquals(expectedException.InnerException.Data);
         }
 
+        private static Expression<Func<Exception, bool>> SameValidationExceptionAs(Exception expectedException)
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.cs:55]: Do we need this? We should be leveraging `SameExceptionAs`

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveAll.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+            this.apiBrokerMock.Verify(broker =>
+                broker.GetAllPostsAsync(),
+                    Times.Once());
+
+            this.apiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+    }
+}
```
Comment (hassanhabib) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveAll.cs:45]: Remove the extra space here brother

Comment (stasnowak) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveAll.cs:45]: @hassanhabib what line exactly, i just can't see it

**Review by ShriHumrudha — APPROVED** [review]

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveAll.cs (hunk 9 above)
Comment (hassanhabib) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveAll.cs:45]: @stasnowak it's gone. weird. maybe I was imagining lol

**Pull request merged**